internal/db/id/collection.go

```diff
@@ -61,10 +76,38 @@ func SetShortCollectionID(
 		return err
 	}
 
-	shortID, err := colSeq.Next(ctx, txn)
+	sID, err := colSeq.Next(ctx, txn)
+	if err != nil {
+		return err
+	}
+	shortID := uint32(sID)
+
+	err = txn.Systemstore().Set(ctx, key.Bytes(), []byte(strconv.Itoa(int(shortID))))
 	if err != nil {
 		return err
 	}
 
-	return txn.Systemstore().Set(ctx, key.Bytes(), []byte(strconv.Itoa(int(shortID))))
+	cache[collectionID] = shortID
+
+	return nil
+}
+
+type collectionShortIDCacheKey struct{}
+
+var collectionShortIDCacheKeyValue = collectionShortIDCacheKey{}
```

Comment:
nitpick: I'm not a fan of this. It's really normal to use the type directly and having the extra variable is a bit confusing and unnecessary.

Reply:
Fair enough, I'll remove it.

- [ ] Remove unnecessary var

Thanks for spotting the JS failure, the job has been very flaky lately and I didn't bother checking this one

- [ ] Fix JS build